docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,131 @@
+
+## How to save record service message
+
+Service introspection ([ros2/ros2#1285](https://github.com/ros2/ros2/issues/1285)) had been implemented. All service messages (requests and responses) are published to service event topic which is hidden topic named as `_service_event` under service name (e.g. /add_two_ints/_service_event). So all messages in this topic can be recorded like general topic.
+
+## How to distinguish service event topic in all recorded topics
+
+Service event topic is saved like general topic. But they should use different process after reading from recording file. So they should be distinguished by topic name.
+
+- General topic  
+  
+  [NameSpace/]TopicName
+
+- Hidden topic  
+  
+  [NameSpace/]_TopicName
+
+- Service event topic
+  
+  [NameSpace/]ServiceName/_service_event
+
+The following criteria will be used to determine a service event topic:
+1. The topic ends with `/_service_event`
+2. The topic type name matches  `*/srv/*_Event` (e.g. `example_interfaces/srv/AddTwoInts_Event`)
+
+### Expand the 'record' command
+
+Add 5 parameters.
+
+- `--services ServiceName1 [ServiceName2 ...]`
+  
+    Specify which services will be recorded. User must specify at least one service.
+
+- `--all-topics`
+
+    Record all topics. Hidden topics including service event topics are excluded.
+
+- `--all-services`
+
+    Record all service event topics.
+
+- `--exclude-topics`
+
+    Rename from `--exclude`. Exclude topics containing provided regular expression.
+
+- `--exclude-services`
+
+    Exclude services containing provided regular expression.
+
+The description of the relevant parameter behavior.
+
+| Parameter | Description |
+| :-- | :--|
+|--include-hidden-topics| Record all hidden topics. Include service event topic. |
+|--services ServiceName1 [ServiceName2 ...] | Record services (service event topics) with specified service names. |
+| --all | Record all topics and service event topics. Other hidden topics are excluded. |
+| --all-topics | Record all topics. Hidden topics including service event topics are excluded. |
+| --all-services | Only record all service event topics. |
+| --exclude-topics | Exclude topics containing provided regular expression. |
+| --exclude-services | Exclude services containing provided regular expression. |
+| -e REGEX, --regex REGEX | Record only topics and service containing provided regular expression. |
+
+### Change output of 'Info' command
+
+A new parameter is added.
+
+- `-v` or `--verbose`
+
+    This parameter only affect service event topic. if `-v` or `--verbose` is set, info command shows the number of request and response for each services based on service event. Otherwise, only show the number of service event. Note that parsing the number of request and response need spent time. The duration of the parsing is related to the number of recorded service events.
```

Comment:
```suggestion
This parameter only affects the service event topic. if `-v` or `--verbose` is set, the info command shows the number of requests and responses for each service based on the service event. Otherwise, only show the number of service events. Note that parsing the number of request and response need spent time. The duration of the parsing is related to the number of recorded service events.
```
Grammar fixes